controllers/gatekeeper_controller.go

```diff
@@ -138,19 +144,20 @@ func (r *GatekeeperReconciler) deployGatekeeperResources() error {
 			return errors.Wrapf(err, "Unable to unmarshal YAML bytes for asset name %s", assetName)
 		}
 
-		if err = r.updateOrCreateResource(manifest); err != nil {
+		if err = r.updateOrCreateResource(manifest, gatekeeper); err != nil {
 			return err
 		}
 	}
 
 	return err
 }
 
-func (r *GatekeeperReconciler) updateOrCreateResource(manifest *manifest.Manifest) error {
+func (r *GatekeeperReconciler) updateOrCreateResource(manifest *manifest.Manifest, gatekeeper *operatorv1alpha1.Gatekeeper) error {
 	ctx := context.Background()
 	clusterObj := &unstructured.Unstructured{}
 	clusterObj.SetAPIVersion(manifest.Obj.GetAPIVersion())
 	clusterObj.SetKind(manifest.Obj.GetKind())
+	ctrl.SetControllerReference(gatekeeper, manifest.Obj, r.Scheme)
```

Comment:
I don't believe this will work for deleting Gatekeeper's cluster-scoped resources e.g. CRDs, validating webhook configuration, cluster role and cluster role binding, because the gatekeeper CR is namespace-scoped and cannot be an owner for cluster-scoped resources.

Reply:
Right, that could be achieved via finalizers. @sabre1041 let me know if you want to add finalizer support to delete the cluster-scoped resources. Else, I could start that work after this PR merges once some details are fixed.

Reply:
@font @JAORMX Added finalizer and addressed remaining issues